protocol/messenger_handler.go

```diff
@@ -1071,19 +1072,42 @@ func (m *Messenger) HandleSendTransaction(messageState *ReceivedMessageState, co
 	if err != nil {
 		return err
 	}
-	transactionToValidate := &TransactionToValidate{
-		MessageID:       messageState.CurrentMessageState.MessageID,
-		CommandID:       command.Id,
-		TransactionHash: command.TransactionHash,
-		FirstSeen:       messageState.CurrentMessageState.WhisperTimestamp,
-		Signature:       command.Signature,
-		Validate:        true,
-		From:            messageState.CurrentMessageState.PublicKey,
-		RetryCount:      0,
+
+	oldMessage, err := m.persistence.MessageByID(command.Id)
+	if err != nil {
+		return err
+	}
+	if oldMessage == nil {
+		return errors.New("message not found")
+	}
```

Comment:
I think it is one or the other. If you have err != nil you oldMessage will not be nil

Reply:
Actually followed what's done in HandleDeclineRequestTransaction. not sure if there is a chance where these two can be both hit?

Reply:
Going deep down into the finder:
```
err = db.tableUserMessagesScanAllFields(row, &message)
	switch err {
	case sql.ErrNoRows:
		return nil, common.ErrRecordNotFound
	case nil:
		return &message, nil
	default:
		return nil, err
	}
```

I think we can assume that.

Reply:
thanks for checking. will do  :)